cmd/config/cmd/merge3_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/cmd/config/cmd"
+	"sigs.k8s.io/kustomize/kyaml/copyutil"
+)
+
+// TestMerge3Command verifies the merge3 correctly applies the diff between 2 sets of resources into another
+func TestMerge3Command(t *testing.T) {
+	datadir, err := ioutil.TempDir("", "test-data")
+	defer os.RemoveAll(datadir)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = ioutil.WriteFile(filepath.Join(datadir, "java-deployment.resource.yaml"), []byte(`apiVersion: apps/v1
```

Comment:
I experienced an extremely strange behavior which I couldn't pinpoint. I found my way around it, but I never resolved the reasoning.

For all the temp files written in this test e.g. all 4 java-deployment.resource.yaml's, whenever I started a file with a new line as such:
```go
err = ioutil.WriteFile(filepath.Join(expected_dir, "java-deployment.resource.yaml"), []byte(`
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app
  labels:
    app: java
    new-local: label
    new-remote: label
```
rather than something like this
```go
err = ioutil.WriteFile(filepath.Join(expected_dir, "java-deployment.resource.yaml"), []byte(`apiVersion: apps/v1
kind: Deployment
metadata:
  name: app
  labels:
    app: java
    new-local: label
    new-remote: label
```
the merge3 command always stripped the newline from the destination file, which resulted in it always being a 1 line diff where the diff was the missing newline. I couldn't tell if this was intended behavior with merge3 rules or if it was a bug with the merge3 library.

Reply:
I think this is an issue in the yaml v3 go libraries.  I've noticed some oddities around how it does comment parsing as well.